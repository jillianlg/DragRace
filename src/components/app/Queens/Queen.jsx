import React from 'react';
import PropTypes from 'prop-types';

const Queen = ({ image_url, name }) => (
  <div>
    <p>{name}</p>
    <img style={{ height: '200px' }} src={image_url} alt={name} />
  </div>
);

Queen.propTypes = {
  image_url: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default Queen;
