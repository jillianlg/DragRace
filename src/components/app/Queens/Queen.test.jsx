import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Queen from './Queen';


describe('Queen component', () => {
  afterEach(() => cleanup());
  it('renders Queen', () => {
    const { asFragment } = render(
      <Queen
        name="Bianca Del Rio"
        image_url="BDR.com"
      />);
    expect(asFragment()).toMatchSnapshot();
  });
});
