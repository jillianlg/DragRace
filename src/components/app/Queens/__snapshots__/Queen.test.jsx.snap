// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queen component renders Queen 1`] = `
<DocumentFragment>
  <div>
    <p>
      Bianca Del Rio
    </p>
    <img
      alt="Bianca Del Rio"
      src="BDR.com"
      style="height: 200px;"
    />
  </div>
</DocumentFragment>
`;
